### ISSUE TITLE
Diablo immortal crashes on startup 

### ISSUE THREAD

**Issue by wed4j** [GrapheneOS/os-issue-tracker#1233]
Hello,
The game "Diablo Immortal" with package name `com.blizzard.diablo.immortal` was released yesterday.
Unfortunately, it crashes on startup. I have looked at all the known issue on other devices, and I have not found anyone with the same problem, so I have to assume the issue is related to graphene.
I have downloaded and installed the game both from aurora and directly as an apk, with the same results.

Here is a logcat of the entire device from the moment just before starting the game, to the moment right after it crashed.

https://gist.githubusercontent.com/wed4j/1ca0369e54c1f2455856bf77c6d03350/raw/e91f27a4eefa40f7c21bd9a61efd30d4ee4cf6ea/gistfile1.txt

I have no experience in Java or Android development, but as I really want to play this game I would be happy to help in any way possible.

Have a good day!

**Comment by Zanthed** [comment]
```
06-02 15:40:17.287 29875 29875 E DeviceInfo: DeviceInfo [getDeviceMD5] Exception1=java.lang.SecurityException: getDeviceId: The uid 10153 does not meet the requirements to access device identifiers.
06-02 15:40:17.287 29875 29875 W System.err: java.lang.SecurityException: getDeviceId: The uid 10153 does not meet the requirements to access device identifiers.
06-02 15:40:17.287 29875 29875 W System.err: 	at android.os.Parcel.createExceptionOrNull(Parcel.java:2426)
06-02 15:40:17.287 29875 29875 W System.err: 	at android.os.Parcel.createException(Parcel.java:2410)
06-02 15:40:17.287 29875 29875 W System.err: 	at android.os.Parcel.readException(Parcel.java:2393)
06-02 15:40:17.287 29875 29875 W System.err: 	at android.os.Parcel.readException(Parcel.java:2335)
06-02 15:40:17.287 29875 29875 W System.err: 	at com.android.internal.telephony.ITelephony$Stub$Proxy.getDeviceIdWithFeature(ITelephony.java:12108)
06-02 15:40:17.287 29875 29875 W System.err: 	at android.telephony.TelephonyManager.getDeviceId(TelephonyManager.java:1972)
```

@muhomorr Maybe something relevant to https://github.com/GrapheneOS/platform_frameworks_base/blob/26c9bfbd93c1e63e0f8c335d06d11797cf2ce7b3/core/java/com/android/internal/gmscompat/sysservice/GmcTelephonyManager.java#L39-L47 ?

**Comment by Gisbitus** [comment]
Bump, same exact issue.

**Comment by Zanthed** [comment]
It's a memory corruption bug that should be reported to the app developers.

`06-02 15:40:35.569   971   971 I Zygote  : Process 29875 exited due to signal 11 (Segmentation fault)`

**Comment by chriswoope** [comment]
Have you investigated it?

While a memory corruption (presumably uncovered by the GrapheneOS allocator) might be the most likely explanation, a segmentation fault could be caused by pretty much anything (e.g. by an API returning null or another value that the app doesn't expect, like the getDeviceId call logged above).

**Comment by thestinger** [comment]
That call fails on the stock OS too and isn't the cause of an issue. It also shouldn't be trying to fetch that identifier and is in violation of Play Store policy.

**Comment by thestinger** [comment]
No, you're mixing up Android privacy improvements with Play Store policy.

**Comment by thestinger** [comment]
> In any case, why does stock Android>10 that also blocks access do not crash? Would it be possible to send a randomized IMEI instead? (If that's really the source of the crash)

It's not the cause of the crash, there is nothing we need to change, and it's off-topic.

**Comment by MrSidi2** [comment]
>  it's off-topic.

Ok, too bad, as I was quite interested in it.

> It's not the cause of the crash

> It's a memory corruption bug that should be reported to the app developers.
> 
> `06-02 15:40:35.569 971 971 I Zygote : Process 29875 exited due to signal 11 (Segmentation fault)`

I missed this, I thought getDeviceId was related to the crash. Thanks for following up

So... `wontfix`? I still find it strange that this particular segfault for this app is only known to occur on Graphene. Is there nothing that can be done?

**Comment by thestinger** [comment]
It's a memory corruption bug in the app and needs to be fixed by them. Read https://grapheneos.org/usage#bugs-uncovered-by-security-features and https://grapheneos.org/features#exploit-protection.
